index.js

```diff
@@ -486,4 +483,31 @@ FeatureService.prototype._catchErrors = function (task, e, url, cb) {
   }.bind(this), task.retry * 1000)
 }
 
+/* Forms a standardized error object for reporting back up the chain
+ * @param {object} error - http request error
+ * @param {string} message - human readable message about ther
+ * @param {object} json - json returned from server request
+ * @returns {object} - error object containing diagnostic information
+ */
+FeatureService.prototype._buildError = function (url, message, error) {
+  if (!error) error = {}
+  error.message = message
+  error.request = url
+  error.response = error.message
+  error.type = error.type || 'Unknown failure'
+  error.code = error.code || 500
+
+  return error
+}
```

Comment:
I asked for some feedback from some people who've been involved with core node development for a long time and got some good advice.

Pertinent parts are here:

https://twitter.com/ungoldman/status/631216971636326400
https://twitter.com/isntitvacant/status/631217038090874880
https://twitter.com/isntitvacant/status/631217716200783873
https://twitter.com/isntitvacant/status/631218312362393600